internal/services/configstore/action/project.go

```diff
@@ -59,6 +59,29 @@ func (h *ActionHandler) ValidateProjectReq(ctx context.Context, req *CreateUpdat
 			return util.NewAPIError(util.ErrBadRequest, errors.Errorf("empty remote repository path"))
 		}
 	}
+
+	err := h.d.Do(ctx, func(tx *sql.Tx) error {
```

Comment:
This shouldn't be done here creating an additional transaction but inside Create and Update project.